cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,259 @@
+package methods
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/creachadair/jrpc2/handler"
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/support/log"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/transactions"
+)
+
+// TransactionsPaginationOptions defines the available options for paginating through transactions.
+type TransactionsPaginationOptions struct {
+	Cursor *transactions.Cursor `json:"cursor,omitempty"`
+	Limit  uint                 `json:"limit,omitempty"`
+}
+
+// GetTransactionsRequest represents the request parameters for fetching transactions within a range of ledgers.
+type GetTransactionsRequest struct {
+	StartLedger uint32                         `json:"startLedger"`
+	EndLedger   uint32                         `json:"endLedger"`
```

Comment:
should EndLedger be optional? the limit will implicitly define the end ledger

Reply:
the limit is on the number of transactions, not ledgers, right? so not necessarily

Reply:
@tamirms Yes, as George pointed out, the limit is number of transactions, not ledgers. And from a user perspective, it would be much easier to just specify the end ledger rather than the API figuring it out for them.

Reply:
@aditya1702 

The `endLedger` filter is only useful if there exists some ledger range where the number of transactions in `[startLedger, endLedger]` is less than `limit`. However, if you take a look at the number of successful + failed transactions included in recent ledgers, https://horizon.stellar.org/ledgers?order=desc , you will find the sum is greater than the default `max-transactions-limit` value (200). So, for pretty much all requests the `limit` parameter will end up restricting how many transactions are included in the response and not the `endLedger` parameter.

Even if there were ledger ranges which had fewer transactions than the requested limit, the client could simply just ignore the transactions in the response which are outside the desired ledger range.

BTW the horizon `/transactions` endpoint does not have an `endLedger` parameter and I'm not aware of any complaints about that being a missing feature.

Some other differences between this endpoint and the horizon `/transactions` endpoint:

- `startLedger` is not necessary because the `startLedger` can be encoded is a toid and included as a cursor
- horizon has an `order` parameter which allows you to return transactions in ascending or descending order which is useful because it provides an easy way to query the newest or oldest transactions

Reply:
We should keep in mind, though, that the purpose of this new endpoint is to facilitate, as @tomerweller put it, "poor man's ingestion." With that in mind, we don't need `order` and can optimize for the forward-moving case, and I think a `startLedger` still makes sense given the fact that TOIDs are too opaque for users to be able to pass along without an SDK to help them. Users won't see a cursor until they make an initial request. Maybe we can do `startLedger xor cursor`?

Reply:
@Shaptic I agree and that is what I have added in this implementation  (similar to `getEvents`) -  on the first request, user will specify a `startLedger` value and then on the subsequent requests they can just use the cursor returned from last response.

Reply:
Without the order parameter it is not possible retrieve the last n ledgers within a single request.

startLedger xor cursor is fine by me. But I’m not seeing the utility of the end ledger parameter

Reply:
I agree, I don't think we need it since it's covered implicitly by `limit`. As for 

> the last n ledgers 

we can add this once it's requested instead of preemptively.

Reply:
Resolving this since we all agreed to remove endLedger in the platform discussion